### PR TITLE
Classic frogger game

### PR DESCRIPTION
**Pull request by NadiaVorontsova**
### Object Oriented JS (classic-frogger-game)

[Demo](https://nadiavorontsova.github.io/classic-frogger-game/) |
[Code base](https://github.com/NadiaVorontsova/classic-frogger-game)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by NadiaVorontsova** [comment]
I finished self-check

File: submissions/NadiaVorontsova/classic-frogger-game/app.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    this.sprite = 'images/enemy-bug.png';
+};
+
+Enemy.prototype.update = function (dt) {
+    this.x += this.speed * dt;
+    if (this.x > ctx.canvas.width) {
+        this.x = startX;
+    }
+    if (this.checkCollision()) {
+        player.resetPlayerPosition();
```
Comment (OleksiyRudenko) [inline submissions/NadiaVorontsova/classic-frogger-game/app.js:74]: >     * [ ] classes do not refer to any global variables, like global variable `player`, which is an instance of `Player` class
>      (referring to global constants and globals provided by the gaming platform like `Resources` is OK);
>      Hint: pass `Player` instance as an argument to every enemy

^^ here and below in the code

File: submissions/NadiaVorontsova/classic-frogger-game/app.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+            countScore--;
+            score();
+        }
+    }
+};
+
+const allEnemies = [];
+
+function createEnemies() {
+    const positionY = [60, 140, 220];
```
Comment (OleksiyRudenko) [inline submissions/NadiaVorontsova/classic-frogger-game/app.js:95]: As a peer developer I am tasked to add more rows to the field and more enemies to the game.
Can these mysterious numbers be calculated from something comprehensible?

File: submissions/NadiaVorontsova/classic-frogger-game/app.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    for (let y of positionY) {
+        const randomPositionX = Math.random() * (-100) - 100;
+        allEnemies.push(new Enemy(randomPositionX, y));
+    }
```
Comment (OleksiyRudenko) [inline submissions/NadiaVorontsova/classic-frogger-game/app.js:99]: A classic use case for `Array#map`

File: submissions/NadiaVorontsova/classic-frogger-game/app.js [hunk 4]
```diff
@@ -0,0 +1,136 @@
+const startPlayerPositionX = 200,
+    startPlayerPositionY = 390,
+    cellWidth = 101,
+    cellHeight = 83,
+    startX = 0,
+    startY = 0,
+    rightWall = 400,
+    bottomWall = 350;
```
Comment (OleksiyRudenko) [inline submissions/NadiaVorontsova/classic-frogger-game/app.js:8]: By convention the global constants names of this kind are ALL_CAPITALS. This way when we see a variable all capitals we instantly understand it is a global constant.

File: submissions/NadiaVorontsova/classic-frogger-game/app.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+        player.resetPlayerPosition();
+    }
+};
+
+Enemy.prototype.render = function () {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Enemy.prototype.checkCollision = function () {
+    if (Math.abs(this.x - player.x) < 50 && Math.abs(this.y - player.y) < 50) {
```
Comment (OleksiyRudenko) [inline submissions/NadiaVorontsova/classic-frogger-game/app.js:83]: What is the semantic meaning of `50`?

File: submissions/NadiaVorontsova/classic-frogger-game/app.js [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+        }
+    }
+};
+
+const allEnemies = [];
+
+function createEnemies() {
+    const positionY = [60, 140, 220];
+    for (let y of positionY) {
+        const randomPositionX = Math.random() * (-100) - 100;
```
Comment (OleksiyRudenko) [inline submissions/NadiaVorontsova/classic-frogger-game/app.js:97]: What is the semantic meaning of `100`?

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@NadiaVorontsova well done!
Although needs improvements.

File: submissions/NadiaVorontsova/classic-frogger-game/app.js (hunk 1 above)
Comment (NadiaVorontsova) [reply submissions/NadiaVorontsova/classic-frogger-game/app.js]: I fixed it.

File: submissions/NadiaVorontsova/classic-frogger-game/app.js (hunk 2 above)
Comment (NadiaVorontsova) [reply submissions/NadiaVorontsova/classic-frogger-game/app.js]: I changed this array

File: submissions/NadiaVorontsova/classic-frogger-game/app.js (hunk 3 above)
Comment (NadiaVorontsova) [reply submissions/NadiaVorontsova/classic-frogger-game/app.js]: +

File: submissions/NadiaVorontsova/classic-frogger-game/app.js (hunk 4 above)
Comment (NadiaVorontsova) [reply submissions/NadiaVorontsova/classic-frogger-game/app.js]: Changed to capital letters

File: submissions/NadiaVorontsova/classic-frogger-game/app.js (hunk 5 above)
Comment (NadiaVorontsova) [reply submissions/NadiaVorontsova/classic-frogger-game/app.js]: It is a size of player. I added a const variable with the size of the player.

File: submissions/NadiaVorontsova/classic-frogger-game/app.js (hunk 6 above)
Comment (NadiaVorontsova) [reply submissions/NadiaVorontsova/classic-frogger-game/app.js]: I removed this variable and passed const START_X.

**Review by OleksiyRudenko — APPROVED** [review]
@NadiaVorontsova great job!

**Comment by NadiaVorontsova** [comment]
> @NadiaVorontsova great job!

Thanks!

**Pull request merged**